### ISSUE TITLE
Running app with User authenticated but empty cache

### ISSUE THREAD

**Issue by andrenpt** [juunas11/aspnetcore2aadauth#12]
When I run the app the first time and I authenticate the user all works as expected.

When I run the app the second time the user is authenticated but the TokenCache it's empty then an exception is raised and it redirects again to the authentication page.

I would believe that if a user starts the application it should fill automatically the TokenCache and also if a user tried to access a controller with the Authorize Attribute it should automatically handle this. 

So basically every time the cache is empty an exception needs to be raised to refill the Token Cache and that does not seem correct !?

Do you have any suggestion to resolve this ?

Thanks in advance
